gnestor/jupyterlab-prodigy
# Migrate build step to be compatible with Jupyterlab>=3.0.0

This PR migrates our extension to be compatible with Jupyterlab >= 3.0.0. Our current version only works on `jupyterlab>=2.0.0, <3.0.0`. 

## How to test

I suggest running the commands in the ["Develop"](https://github.com/ljvmiranda921/jupyterlab-prodigy/tree/migrate-to-v3#-develop) section **or** follow the production checks in [the Github action](https://github.com/ljvmiranda921/jupyterlab-prodigy/blob/migrate-to-v3/.github/workflows/build.yml). Please ensure that **your jupyterlab version is >=3.0.0**.

You can also start a sample Prodigy session by running:

```bash
prodigy ner.manual my_set blank:en notebooks/news_headlines.jsonl --label PERSON,ORG,PRODUCT
```

If something fails, please let me know:
- Your jupyterlab version (`jupyter --version`)
- Your prodigy version
- Traceback, if possible.

## Next steps

After this PR is merged, we need to publish this both in npm and PyPI (yes, we need to publish it in PyPI as well). Here's a checklist of what we should do:

- [x] Bump version using `npm version` (I recommend a `major` version change. This is **not compatible** with older versions of jupyterlab)
- [ ] Publish package in npm 
- [ ] Publish package in PyPI (the package version is obtained from `package.json` already, so no need to bump any `.py` files).
- [ ] Announce? Quick tweet? or just in Prodigy support?
